rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1132,6 +1132,40 @@ public Sha256Hash getBtcBlockchainBlockHashAtDepth(int depth) throws BlockStoreE
         return current.getHeader().getHash();
     }
 
+    public Long getBtcTransactionConfirmationsGetCost(Sha256Hash btcBlockHash) {
+        final long BASIC_COST = 13_600;
+        final long STEP_COST = 70;
+
+        // Dynamic cost based on the depth of the block that contains
+        // the transaction. Find such depth first, then calculate
+        // the cost.
+        Context.propagate(btcContext);
+        try {
+            this.ensureBtcBlockStore();
+            final StoredBlock block = btcBlockStore.getFromCache(btcBlockHash);
+
+            // Block not found, default to basic cost
+            if (block == null) {
+                return BASIC_COST;
+            }
+
+            final int bestChainHeight = getBtcBlockchainBestChainHeight();
+            final int blockDepth = bestChainHeight - block.getHeight();
+
+            // Block too deep, default to basic cost
+            if (blockDepth > BTC_TRANSACTION_CONFIRMATION_MAX_DEPTH) {
```

Comment:
Retrieving old blocks from disk (cache is limited to 65k items) is cheaper than traversing BTC_TRANSACTION_CONFIRMATION_MAX_DEPTH blocks in RAM. Is this correct?

Reply:
I'm not entirely sure on what do you mean or where does this question lead. Regardless:

- Retrieving one single block from disk is probably cheaper than retrieving BTC_TRANSACTION_CONFIRMATION_MAX_DEPTH (4320 at the time of writing) blocks from RAM, but I would have to check and get back to you. 
- Retrieving a given number of blocks from RAM should be cheaper than retrieving that same number of blocks from disk.